src/backend/gl/src/command.rs

```diff
@@ -326,44 +327,50 @@ impl RawCommandBuffer {
 
     fn update_blend_targets(&mut self, blend_targets: &Vec<pso::ColorBlendDesc>) {
         let max_blend_slots = blend_targets.len();
+        if max_blend_slots == 0 {
+            return;
+        }
 
-        if max_blend_slots > 0 {
-            match self.cache.blend_targets {
-                Some(ref mut cached) => {
-                    if cached.len() < max_blend_slots {
-                        cached.resize(max_blend_slots, None);
-                    }
-                }
-                None => {
-                    self.cache.blend_targets = Some(vec![None; max_blend_slots]);
-                }
-            };
+        if self.cache.blend_targets.len() < max_blend_slots {
+            self.cache.blend_targets.resize(max_blend_slots, None);
         }
 
-        for (slot, blend_target) in blend_targets.iter().enumerate() {
+        let mut all_targets_same = true;
+        for blend_target in &blend_targets[1..] {
+            if blend_target != &blend_targets[0] {
+                all_targets_same = false;
+                break;
+            }
+        }
+
+        if all_targets_same {
             let mut update_blend = false;
-            if let Some(ref mut cached_targets) = self.cache.blend_targets {
-                if let Some(cached_target) = cached_targets.get(slot) {
-                    match cached_target {
-                        &Some(ref cache) => {
-                            if cache != blend_target {
-                                update_blend = true;
-                            }
-                        }
-                        &None => {
-                            update_blend = true;
-                        }
-                    }
+            for cached_target in &mut self.cache.blend_targets {
+                if cached_target.as_ref() != Some(&blend_targets[0]) {
```

Comment:
*theoretically*, this could cache the case where there's more render buffers now than there were before, but before settings were changed across all render buffers so we wouldn't need to set the settings on the new render buffers to match it.  It does not do that currently.